scispacy/umls_utils.py

```diff
@@ -36,49 +36,75 @@ def __repr__(self):
         return rep
 
 
+# Keep this for backward compatability.
+UmlsEntity = Entity
+
+
 DEFAULT_UMLS_PATH = (
     "https://s3-us-west-2.amazonaws.com/ai2-s2-scispacy/data/umls_2017_aa_cat0129.json"
 )
 DEFAULT_UMLS_TYPES_PATH = "https://s3-us-west-2.amazonaws.com/ai2-s2-scispacy/data/umls_semantic_type_tree.tsv"
 
 
-class UmlsKnowledgeBase:
-
+class KnowledgeBase:
     """
-    A class representing two commonly needed views of the Unified Medical Language System:
-    1. A mapping from concept_id to a UmlsEntity NamedTuple with more information.
+    A class representing two commonly needed views of a Knowledge Base:
+    1. A mapping from concept_id to an Entity NamedTuple with more information.
     2. A mapping from aliases to the sets of concept ids for which they are aliases.
 
     Parameters
     ----------
-    file_path: str, optional.
-        The file path to the json representation of UMLS to load.
-
+    file_path: str, required.
+        The file path to the json/jsonl representation of the KB to load.
     """
 
     def __init__(
-        self,
-        file_path: str = DEFAULT_UMLS_PATH,
-        types_file_path: str = DEFAULT_UMLS_TYPES_PATH,
+        self, file_path: str = None,
     ):
-        raw = json.load(open(cached_path(file_path)))
+        if file_path is None:
+            raise ValueError(
+                "Do not use the default arguments to KnowledgeBase. "
+                "Instead, use a subclass (e.g UmlsKnowledgeBase) or pass a path to a kb."
+            )
+        if file_path.endswith("jsonl"):
+            raw = [json.loads(line) for line in open(cached_path(file_path))]
+        else:
+            raw = json.load(open(cached_path(file_path)))
 
         alias_to_cuis: Dict[str, Set[str]] = defaultdict(set)
-        self.cui_to_entity: Dict[str, UmlsEntity] = {}
+        self.cui_to_entity: Dict[str, Entity] = {}
 
         for concept in raw:
             unique_aliases = set(concept["aliases"])
             unique_aliases.add(concept["canonical_name"])
             for alias in unique_aliases:
                 alias_to_cuis[alias].add(concept["concept_id"])
-            self.cui_to_entity[concept["concept_id"]] = UmlsEntity(**concept)
+            self.cui_to_entity[concept["concept_id"]] = Entity(**concept)
 
         self.alias_to_cuis: Dict[str, Set[str]] = {**alias_to_cuis}
+
+
+class UmlsKnowledgeBase(KnowledgeBase):
+    def __init__(
+        self,
+        file_path: str = DEFAULT_UMLS_PATH,
+        types_file_path: str = DEFAULT_UMLS_TYPES_PATH,
+    ):
+
+        super().__init__(file_path)
+
         self.semantic_type_tree: UmlsSemanticTypeTree = construct_umls_tree_from_tsv(
             types_file_path
         )
 
 
+class MeshKnowledgeBase(KnowledgeBase):
```

Comment:
I believe mesh terms have a tree as well, if one of us is able to find that tree somewhere on the internet it would be nice to package it here.

Reply:
They do, but the only reason we have the tree in there in the first place was because we were messing around with changing the granularity of the types for the NER model, which turned out to not be super useful. Also, the MESH KB lives in this hyper complex sparql database and I really don't want to figure out how to extract the tree haha

Reply:
lol fair enough. i do think that one of the pieces of value we add here is easy access to these kbs that are pretty complicated to figure out how to access (and people do use the type tree, or at least the levels of the types). so if its not too much work, it'd be great to extract and add, if its difficult, im fine letting it go.

Reply:
That's true - I'll add a TODO, maybe we can do it later. Mesh has a much better browser for viewing the entities, e.g https://meshb.nlm.nih.gov/record/ui?ui=D002430

Reply:
maybe just open an issue, and if someone has a compelling use case we can add it, or maybe they will add it :)